lib/espresso-runner.js

```diff
@@ -17,31 +23,56 @@ class EspressoRunner {
     }
     this.jwproxy = new JWProxy({host: this.host, port: this.systemPort});
     this.proxyReqRes = this.jwproxy.proxyReqRes.bind(this.jwproxy);
+
+    this.modServerPath = path.resolve(this.tmpDir, `${TEST_APK_PKG}.apk`);
   }
 
   // TODO mostly duplicated from uiautomator2's installServerApk
   async installTestApk () {
-    // Installs the apks on to the device or emulator
-    let isApkInstalled = await this.adb.isAppInstalled(TEST_APK_PKG);
-    if (isApkInstalled) {
-      //check server apk versionName
-      let apkVersion = await this.getAPKVersion(TEST_APK_PATH);
-      let pkgVersion = await this.getInstalledPackageVersion(TEST_APK_PKG);
-      if (apkVersion !== pkgVersion) {
-        isApkInstalled = false;
-        await this.adb.uninstallApk(TEST_APK_PKG);
+    for (let pkg of [TEST_APK_PKG, APP_APK_PKG]) {
+      if (await this.adb.isAppInstalled(pkg)) {
+        // for now, uninstall always
+        // TODO: figure out how to know if the server is able to instrument the AUT
+        await this.adb.uninstallApk(pkg);
       }
     }
-    if (!isApkInstalled) {
-      await this.signAndInstall(TEST_APK_PATH, TEST_APK_PKG);
+
+    let tmpTestApkPath = await this.buildNewModServer();
+    await this.signAndInstall(this.modServerPath, TEST_APK_PKG);
+
+    await this.signAndInstall(APP_APK_PATH, APP_APK_PKG);
+  }
+
+  async buildNewModServer () {
```

Comment:
That is what my comment on line 33 is about. This is all just to get a prototype working.